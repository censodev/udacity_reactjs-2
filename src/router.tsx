import { createBrowserRouter } from "react-router-dom";
import Login from "./pages/Login";
import MainLayout from "./components/MainLayout";
import Home from "./pages/Home";
import Leaderboard from "./pages/Leaderboard";
import QuestNew from "./pages/QuestNew";
import QuestDetail from "./pages/QuestDetail";
const router = createBrowserRouter([
    {
        path: '/login',
        element: <Login />,
    },
    {
        path: '/',
        element: <MainLayout />,
        children: [
            {
                path: '/',
                element: <Home />,
            },
            {
                path: '/leaderboard',
                element: <Leaderboard />,
            },
            {
                path: '/add',
                element: <QuestNew />,
            },
            {
                path: '/quest/:id',
                element: <QuestDetail />,
            },
        ]
    },
])
export default router