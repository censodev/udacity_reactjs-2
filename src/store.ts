import { configureStore } from "@reduxjs/toolkit";
import authReducer from './slices/authSlice';
import questReducer from './slices/questSlice';

const store = configureStore({
    reducer: {
        auth: authReducer,
        quest: questReducer,
    },
    middleware: getDefaultMiddleware =>
        getDefaultMiddleware({
            serializableCheck: false,
        })
})
export default store
export type RootState = ReturnType<typeof store.getState>