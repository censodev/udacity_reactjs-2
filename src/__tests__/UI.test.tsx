import { act, fireEvent, render, screen, waitFor } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../store';
import { RouterProvider } from 'react-router-dom';
import router from '../router';

describe("UI", () => {
    it("snapshot", async () => {
        const component = render(
            <Provider store={store}>
                <RouterProvider router={router} />
            </Provider>
        );
        expect(component).toMatchSnapshot();
    });
    it("dom load", async () => {
        await act(() => render(
            <Provider store={store}>
                <RouterProvider router={router} />
            </Provider>
        ));
        const loginBtn = screen.getByTestId("btnLogin");
        expect(loginBtn).toBeInTheDocument()
    });
    it("login flow - success", async () => {
        await act(() => render(
            <Provider store={store}>
                <RouterProvider router={router} />
            </Provider>
        ));

        await act(async () => {
            fireEvent.change(screen.getByTestId('username'), { target: { value: 'sarahedo' } })
            fireEvent.change(screen.getByTestId('password'), { target: { value: 'password123' } })
        });

        expect(screen.getByTestId('username')).toHaveValue('sarahedo')
        expect(screen.getByTestId('password')).toHaveValue('password123')

        await act(async () => {
            fireEvent.click(screen.getByTestId("btnLogin"))
        })

        await waitFor(() => {
            expect(screen.getByTestId("home")).toBeInTheDocument()
        }, { timeout: 5000 });

        expect(screen.getByTestId("home")).toBeInTheDocument()
    });
});