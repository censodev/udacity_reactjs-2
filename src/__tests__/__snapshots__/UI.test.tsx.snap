// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`5. Snapshot test tests NewQuestion page with snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        style="padding: .5rem; max-width: 100%;"
      >
        <div
          style="display: flex; align-items: center; gap: .5rem;"
        >
          <strong
            style="font-size: 1rem;"
          >
            Something went wrong!
          </strong>
          <button
            style="appearance: none; font-size: .6em; border: 1px solid currentcolor; padding: .1rem .2rem; font-weight: bold; border-radius: .25rem;"
          >
            Hide Error
          </button>
        </div>
        <div
          style="height: .25rem;"
        />
        <div>
          <pre
            style="font-size: .7em; border: 1px solid red; border-radius: .25rem; padding: .3rem; color: red; overflow: auto;"
          >
            <code>
              window.matchMedia is not a function
            </code>
          </pre>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      style="padding: .5rem; max-width: 100%;"
    >
      <div
        style="display: flex; align-items: center; gap: .5rem;"
      >
        <strong
          style="font-size: 1rem;"
        >
          Something went wrong!
        </strong>
        <button
          style="appearance: none; font-size: .6em; border: 1px solid currentcolor; padding: .1rem .2rem; font-weight: bold; border-radius: .25rem;"
        >
          Hide Error
        </button>
      </div>
      <div
        style="height: .25rem;"
      />
      <div>
        <pre
          style="font-size: .7em; border: 1px solid red; border-radius: .25rem; padding: .3rem; color: red; overflow: auto;"
        >
          <code>
            window.matchMedia is not a function
          </code>
        </pre>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
